mantidproject/mantid
# Extending MagneticIon to 5d ions and including the dipole formula for magnetic form factors

This PR adds the tables of coefficients for 5d ions (Hf-Au) for the analytical approximation to the Hartree-Fock calculated radial integrals going into the magnetic form factor.

In addition, it also amends the method `analyticalFormFactor` to output the form factor F(Q) in the dipole approximation rather than just the spin-only forms. This will only affects rare earth ions (Ce-Yb, U-Am). Transition metal ions will be the same a before.

**To test:**

The changes are not exposed to Python. The easiest way to test from Python is to use the `MagFormFactorCorrection` algorithm from #15703 and specify a name for the `FormFactorWorkspace` property. The algorithm will then produce a point data workspace with the calculated form factor for the Q-range of the input workspace. You can then check that at Q=0, the form factor is 1 and that it falls off to zero by around 20 Angstrom^-1. The analytical approximation is not perfect and the analytical form factor may become negative at some points. 



Fixes #17457 .

Documentation to follow.



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
